src/Parser/Shaman/Elemental/Modules/Items/EchoesOfTheGreatSundering.js

```diff
@@ -6,66 +6,118 @@ import ITEMS from 'common/ITEMS';
+  buffedCastCounter=0;
 
-    on_initialized() {
-        this.active = this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id);
-    }
+  state = 0;  //0=guaranteed not buffed; 1=guaranteed buffed; 2=not too sure(use heuristic)
+  endtime = 0;
 
-    on_byPlayer_cast(event) {
-        const spellId = event.ability.guid;
-        if (spellId !== SPELLS.EARTHQUAKE.id) {
-            return;
-        }
-        if (this.combatants.selected.hasBuff(SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id, event.timestamp)) {
-            this.buffedEarthquakeCasts += 1;
-        }
-        this.totalEarthquakeCasts += 1;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id);
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id) {
+      this.echoesProcsCounter += 1;
     }
+  }
 
-    on_byPlayer_damage(event) {
-        const spellId = event.ability.guid;
-        if (spellId === SPELLS.EARTHQUAKE_DAMAGE.id) {
-            this.totalEarthquakeDamage += event.amount;
-        }
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.EARTHQUAKE.id)
+      return;
+
+    if (this.combatants.selected.hasBuff(SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id, event.timestamp)) {
+      this.buffedCastCounter++;
+      this.endtime = event.timestamp + ECHOES.PROC_DURATION + ECHOES.LAG_TOLERANCE;
+      this.state = 1;
     }
+    else {
+      if (event.timestamp < this.endtime) {
+        this.state = 2;
+      }
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    if (event.timestamp > this.endtime)
+      this.state = 0;
+
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.EARTHQUAKE_DAMAGE.id)
+      return;
 
-    on_byPlayer_applybuff(event) {
-        const spellId = event.ability.guid;
-        if (spellId === SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id) {
-            this.echoesProcsCounter += 1;
+    const critMultiplier = (event.hitType === HIT_TYPES.CRIT) ? CRIT_MULTIPLIER : 1;
+    const baseDamage = event.amount / critMultiplier;
+
+    switch (this.state) {
+      case 0:
+        this.unbuffedBaseDamageSum += baseDamage;
+        this.unbuffedTickCounter++;
+        break;
+      case  1:
+        this.buffedEarthquakeDamage += event.amount;
+        this.buffedBaseDamageSum += baseDamage;
+        this.buffedTickCounter++;
+        break;
+      case 2:
+        if (this.buffedBaseDamageSum !== 0 && this.buffedTickCounter > 0) {
+          const averageBuffedBaseDamage = this.buffedBaseDamageSum / this.buffedTickCounter;
+          if (Math.abs(baseDamage / averageBuffedBaseDamage) < ECHOES.TRESHOLD_PERCENTAGE) {
+            this.buffedEarthquakeDamage += event.amount;
+            this.buffedBaseDamageSum += baseDamage;
+            this.buffedTickCounter++;
+            console.log(this.buffedBaseDamageSum);
```

Comment:
Please do not leave `console.log` statements active in production code. A common practice is to add a `debug` constant at the top of the file that toggles them on or off.